src/WebJobs.Extensions.OpenAI/Embeddings/EmbeddingsAttribute.cs

```diff
@@ -89,57 +99,82 @@ TextReader GetTextReader()
         TextReader reader,
         int minChunkSize,
         int maxChunkSize,
-        char[]? terminatorChars = null)
+        int overlap,
+        char[]? sentenceEndings = null,
+        char[]? wordBreaks = null)
     {
-        char[] buffer = new char[maxChunkSize];
-        int startIndex = 0;
+        if (reader == null)
+        {
+            throw new ArgumentNullException("reader");
+        }
 
-        if (terminatorChars == null)
+        if (minChunkSize < 0 || maxChunkSize <= 0 || overlap < 0 || minChunkSize > maxChunkSize || overlap > maxChunkSize)
         {
-            terminatorChars = new[] { '\n' };
+            throw new ArgumentException("Invalid chunk size or overlap");
         }
 
+        char[] buffer = new char[maxChunkSize];
+        int startIndex = 0;
+
+        sentenceEndings ??= new[] { '.', '!', '?' };
+        wordBreaks ??= new[] { ',', ';', ':', ' ', '(', ')', '[', ']', '{', '}', '\t', '\n' };
```

Comment:
Place these default sentence ending and work break character arrays into `static readonly` variables so that we don't allocate the same data on each call.

Reply:
done